src/engine/pagination.ts

```diff
@@ -0,0 +1,93 @@
+import { queryValueToIntOrUndefined } from '../utils'
+import {
+  Query,
+  AbstractQuery,
+  ParseQuery,
+  useAbstractQuery,
+  Router,
+  AbstractQueryOptions,
+} from './query'
+
+export type PaginationQuery = { page: number; size: number }
+
+export type ChangeQueryOptions = { resetPage: boolean }
```

Comment:
Could you add comments to describe the values. I think especially for `resetPage`  it would not hurt and might not be immediately clear from the name. It would also be nice to see the default value for page, size and resetPage.

Reply:
Please have a look at what I added. I think it is explained now.